packages/proxy/lib/http/index.ts

```diff
@@ -456,15 +465,25 @@ export class Http {
   }
 
   updateServiceWorkerRegistrations (data: Protocol.ServiceWorker.WorkerRegistrationUpdatedEvent) {
-    this.preRequests.updateServiceWorkerRegistrations(data)
+    data.registrations.forEach((registration) => {
```

Comment:
sorry - curious again, why not let the server worker manager handle iteration and messaging this data?

Reply:
I think I was mainly just trying to keep the ServiceWorker sort of pure and not dependent on CDP API specifics. I could go either way on passing that logic one layer deep though I guess.

Reply:
My thought would be that it would contains all the of the service work data/logic expectations in one file and test coverage vs needing to trace it through out.

Reply:
Updated: [`bf92f82` (#28517)](https://github.com/cypress-io/cypress/pull/28517/commits/bf92f825c87e6502deb70ce56631877b39c758b4)